### DISCUSSION TITLE
Mapping custom transforms with particular routes

### DISCUSSION THREAD

**Discussion by mfwojciechowski** [dotnet/yarp#2034] [Q&A] [answered]
Good morning everyone!
I noticed that we can configure specific route to use build-in transforms. I'm wondering, is it possible to create custom transform, using .AddTransform() and map it with a particular routes? That I could use created custom transform let say two routes from all my defined routes? For now it is applied for all defined routes. 

Thanks for help!

**Comment by Tratcher** [comment, marked as answer]
Yes, see https://microsoft.github.io/reverse-proxy/articles/transforms.html#itransformprovider and ITransformFactory

**Comment by MihaZupan** [comment]
Note that the callback you provide in `AddTransforms` is called for every route, so you can apply different logic per route.
```c#
.AddTransforms(builder =>
{
    if (builder.Route.RouteId == "myRouteId")
    {
        builder.AddResponseTransform(async context =>
        {
            // ...
        });
    }
});
```
